### PR TITLE
#9812: Remove deprecation warning in TestAppDelegate

### LINKED ISSUE #9812: Fix Key window warning in TestAppDelegate
Fix Key window warning in TestAppDelegate on line `defer { application.keyWindow?.layer.speed = 100.0 }`

Screen Shot 2022-01-19 at 4 30 30 PM

We can probably use `UIWindow.keyWindow` for it, but needs to be tested.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3644)

### PR DESCRIPTION
**Pull request by noloman**
Closes #9812

### CODE REVIEW

File: Client/Application/TestAppDelegate.swift [hunk 1]
```diff
@@ -173,7 +173,11 @@ class TestAppDelegate: AppDelegate {
 
     override func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Speed up the animations to 100 times as fast.
-        defer { application.keyWindow?.layer.speed = 100.0 }
+        defer {
+            if let keyWindow = UIWindow.key {
```
Comment (lmarceau) [inline Client/Application/TestAppDelegate.swift:177]: Using `UIWindow.keyWindow` it wasn't working? [(link)](https://github.com/mozilla-mobile/firefox-ios/blob/31e22aa0d7fe94629ebaba991e707fc917b0e1e3/Client/UIWindowExtensions.swift#L8)

Comment (noloman) [reply Client/Application/TestAppDelegate.swift]: yes, indeed! Sorry, I didn't realized that that was already there 😄

**Comment by lmarceau** [comment]
Heya @noloman ! We see some extra changes now on this PR  🤔
Screen Shot 2022-02-04 at 9 37 47 AM

**Comment by noloman** [comment]
> Heya @noloman ! We see some extra changes now on this PR  🤔 Screen Shot 2022-02-04 at 9 37 47 AM

Done! let's see if Bitrise works...

**Review by lmarceau — APPROVED** [review]
Thanks @noloman !

**Comment by lmarceau** [comment]
Doubled checked that tests passes locally!

Screen Shot 2022-02-07 at 6 41 28 PM

**Pull request merged**